File: csharp/triangle/Triangle.cs
public static class Triangle
{
    public static bool IsScalene(double x, double y, double z) =>
        IsValid(x, y, z) && (x != y && y != z && x != z);

    public static bool IsIsosceles(double x, double y, double z) =>
        IsValid(x, y, z) && (x == y || y == z || x == z);

    public static bool IsEquilateral(double x, double y, double z) =>
        IsValid(x, y, z) && x == y && y == z;

    private static bool IsValid(double x, double y, double z) =>
        x != 0 && y != 0 && z != 0 &&
        x + y >= z && x + z >= y && y + z >= x;
}

File: nim/triangle/triangle.nim
import algorithm, math, sequtils

type
  Triangle = array[3, int]

proc isValid(triangle: Triangle): bool =
  var triangle = triangle.sorted
  triangle.sum > 0 and triangle[0] + triangle[1] >=  + triangle[2]

proc uniqueSides(triangle: Triangle): int =
  triangle.deduplicate.len

proc isEquilateral*(triangle: Triangle): bool =
  triangle.isValid and triangle.uniqueSides == 1

proc isIsosceles*(triangle: Triangle): bool =
  triangle.isValid and triangle.uniqueSides <= 2

proc isScalene*(triangle: Triangle): bool =
  triangle.isValid and triangle.uniqueSides == 3
